platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbCompactionCoordinator.java

```diff
@@ -0,0 +1,282 @@
+                }
+            });
+            compactionFuturesByName.put(task.id, future);
+        }
+    }
+
+    /**
+     * Creates a thread pool for compaction tasks if it does not exist yet, or returns the existing one.
+     * If the existing thread pool is shutdown, it will be replaced with a new one. Note that there is only one
+     * compacting thread-pool per JVM.
+     * @return a thread pool for compaction tasks
+     */
+    ExecutorService createOrGetCompactingExecutor() {
+        ExecutorService executorService = compactionExecutorServiceRef.get();
+        if (executorService == null) {
+            executorService = new ThreadPoolExecutor(
+                    getCompactingThreadNumber(),
+                    getCompactingThreadNumber(),
+                    0L,
+                    TimeUnit.MILLISECONDS,
+                    new LinkedBlockingDeque<>(),
+                    new ThreadConfiguration(getStaticThreadManager())
+                            .setThreadGroup(new ThreadGroup("Compaction"))
+                            .setComponent(MERKLEDB_COMPONENT)
+                            .setThreadName("Compacting")
+                            .setExceptionHandler((t, ex) ->
+                                    logger.error(EXCEPTION.getMarker(), "Uncaught exception during merging", ex))
+                            .buildFactory());
+            if (!compactionExecutorServiceRef.compareAndSet(null, executorService)) {
+                try {
+                    executorService.shutdown();
+                } catch (Exception e) {
+                    logger.error(EXCEPTION.getMarker(), "Failed to shutdown compaction executor service", e);
+                }
+            }
+        } else {
+            if (executorService.isShutdown()) {
+                compactionExecutorServiceRef.compareAndSet(executorService, null);
+                return createOrGetCompactingExecutor();
+            }
+        }
+
+        return compactionExecutorServiceRef.get();
+    }
+
+    int getCompactingThreadNumber() {
+        return config.compactionThreads();
+    }
+
+    boolean isCompactionEnabled() {
+        return compactionEnabled.get();
+    }
+
+    /**
+     * A helper class representing a task to run compaction for a specific storage type.
+     */
+    private abstract static class CompactionTask implements Callable<Boolean> {
+
+        private static final Logger logger = LogManager.getLogger(CompactionTask.class);
+        final String id;
+
+        public CompactionTask(String id) {
+            requireNonNull(id);
+            this.id = id;
+        }
+
+        protected abstract boolean doCompaction() throws IOException, InterruptedException;
+
+        @Override
+        public Boolean call() {
+            try {
+                return doCompaction();
+            } catch (final InterruptedException | ClosedByInterruptException e) {
+                Thread.currentThread().interrupt();
```

Comment:
If we're expecting to interrupt this `Callable` (e.g. via the interruptible completable future cancel method), we can leave out this line.

Reply:
I removed the `interrupt` call.